doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,218 @@
+
+ - RDM: 1
+ - Title: RIOT design goals
+ - Author: Daniel Petry  
+ - Date: October 2018
+
+## Abstract
+
+This memo describes RIOT's high level design goals. The aim is to reflect
+community consensus on these goals and thereby guide technical discussion around
+specific contributions.
+
+## Status
+
+This document is a product of the community of RIOT maintainers, and aims to 
+represent the consensus within this community. The content of this document is
+licensed with a Creative Commons CC-BY-SA license.
+
+## Introduction
+
+The primary aim of this document is to guide lower level debate and discussion.
+The problems that aim to be solved are that design priorities are currently:
+
+- Expressed as a list of points in the README.md and RIOT vision, without
+  expanding on any of them individually, which gives little understanding of
+  each point
+- The design priorities are expressed as equals in those documents, making it
+  difficult to make design decisions around them where there are tradeoffs
+- Not captured in a place which explicitly focuses on users, which means that
+  they don't foster an intuition of user needs, and therefore of how to make
+  design decisions which lead to RIOT being a widely used product.
+- Otherwise communicated via word-of-mouth, which is inefficient and leads to
+  corruption of information
+- Percieved differently by different people.
+
+In other words, this document aims to increase alignment and visibility of
+information on RIOT's design goals, and to increase their focus on users.
+
+This information is illustrated via several angles:
+
+1. A short mission statement, suitable to gather a 30 second impression of the
+   operating system on a technical level, which can also be re-used in public
+   documents
+2. A series of example use cases
+3. A series of specific design philosophies.
+
+
+# 1. Mission Statement
+
+RIOT is an open-source operating system which makes it easy to deploy sensor and
+actuator networks with challenging resource requirements that can operate for
+years without physical intervention.
+
+It is designed to abstract away the hardware details of remote and low-access
+sensor and actuator nodes. An identical programming interface is presented
+across heterogeneous nodes, supporting underlying hardware out-of-the-box, with
+a short learning curve. Those nodes are often highly constrained in power and
+memory and failures cannot be tolerated due to their isolation, and the
+operating system is designed to handle these constraints with fault-free long
+term operation. The protocols used to communicate with them are at the leading
+edge of innovation in constrained and mesh networking, to ensure data is
+reliably delivered between nodes and to servers.
+
```

Comment:
Should add remark on versatility, e.g., 
"The use cases for IoT systems are manifold and so are their 
requirements. Therefore, design decisions in RIOT should not prefer one technology or 
one protocol over another." (originally noted by @OlegHahm)